### DISCUSSION TITLE
Dimensional Lumber Standard Length Input Question

### DISCUSSION THREAD

**Discussion by DylanStracahn** [lairdubois/lairdubois-opencutlist-sketchup-extension#534] [General]
When using dimensional lumber, you generally use a variety of sizes. Is there any way to have the cutlist generate parts on multiple lengths of material? For instance, you may use 2x4x8 lumber on the side walls of a house, but you will use much longer 2x4s on the gable end. The only workaround I can see is to use multiple textures, but that is quite the chore and highly confusing on larger projects. I'd love for opencutlist to be able to optimize parts across multiple standard lengths to minimize waste/account for varying lengths of parts.

**Comment by mobilarte** [comment]
Hi Dylan,

OpenCutList does not directly work with multiple standard lengths of material. There are many reasons why this is not possible. It would be very complex to define the target function to minimize, which would depend on price per unit length of material (longer material is usually more expensive), handling costs (is handling longer boards easier or more complicated? depends on the crosscutting saw setup), variety of material (is a single length easier to handle than multiple lengths?). If we do not consider all these parameters, the optimization result will be very hard to interpret. Having the user enter all this information will be more confusing.

If you work in construction it should not be too difficult to mark (with tags or badges in v5.0) all your 9' framing studs which will not have to be cut to length anyway. In practice shorter parts like cripples, window headers and sills will more likely fill longer boards than shorter boards. Boards for the gable will need to be longer, but in general plates and rafters too. Offcuts can be used for gable walls.

Have you tried to estimate the difference in price on a large project when using different lengths of boards? How large in percentage of material cost is the difference? How large in percentage of the overall project cost is the difference?

**Reply by DylanStracahn** [reply]
The Tags are an interested approach. I am using this much more for an estimation than for a "cutlist", and the savings may be intangible for the many. If I am going to spend the time to model a project in detail, it would be nice to also benefit form material optimization though.
